yt_dlp/extractor/rtvs.py

```diff
@@ -54,45 +55,35 @@ class RTVSIE(InfoExtractor):
         }
     }]
 
-# TODO
-# live video https://www.rtvs.sk/televizia/live-3 , https://www.rtvs.sk/televizia/sport
-# live radio https://slovensko.rtvs.sk/player
-
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
         webpage = self._download_webpage(url, video_id)
         iframe_id = self._search_regex(
-            r'<iframe[^>]+id\s*=\s*"player_[^_]+_([0-9]+)"', webpage,
-            'Iframe ID')
+            r'<iframe[^>]+id\s*=\s*"player_[^_]+_([0-9]+)"', webpage, 'Iframe ID')
         iframe_url = self._search_regex(
-            fr'<iframe[^>]+id\s*=\s*"player_[^_]+_{iframe_id}"[^>]+src\s*=\s*"([^"]+)"', webpage,
-            'Iframe URL')
+            fr'<iframe[^>]+id\s*=\s*"player_[^_]+_{re.escape(iframe_id)}"[^>]+src\s*=\s*"([^"]+)"', webpage, 'Iframe URL')
 
         webpage = self._download_webpage(iframe_url, video_id, 'Downloading iframe')
-        json_url = self._search_regex(
-            r'var\s+url\s*=\s*"([^"]+)"\s*\+\s*ruurl', webpage,
-            'Json URL')
-        full_json_url = f'https:{json_url}b=mozilla&p=win&v=97&f=0&d=1'
-        data = self._download_json(
-            full_json_url, video_id, 'Downloading json')
+        json_url = self._search_regex(r'var\s+url\s*=\s*"([^"]+)"\s*\+\s*ruurl', webpage, 'json URL')
+        data = self._download_json(f'https:{json_url}b=mozilla&p=win&v=97&f=0&d=1', video_id)
 
         if data.get('clip'):
             data['playlist'] = [data['clip']]
 
-        to_return = {
+        if traverse_obj(data, ('playlist', 0, 'sources', 0, 'type')) == 'audio/mp3':
+            formats = [{'url': traverse_obj(data, ('playlist', 0, 'sources', 0, 'src'))}]
+        else:
+            formats = self._extract_m3u8_formats(traverse_obj(data, ('playlist', 0, 'sources', 0, 'src')), video_id)
+            self._sort_formats(formats)
```

Comment:
```suggestion
        self._sort_formats(formats)
```
This must be always run